web/src/message_list_data.ts

```diff
@@ -221,6 +221,12 @@ export class MessageListData {
             if (message.type !== "stream") {
                 return true;
             }
+
+            // TODO: Widen this if check to include other narrows as well.
+            if (this.filter.is_in_home()) {
+                return user_topics.is_topic_visible_in_home(message.stream_id, message.topic);
+            }
+
```

Comment:
Also, we will need to a query in the backend to support this.

Reply:
Yeah I think we need to have a conversation about this. It might be that this is the wrong direction, and we want to just exclude the `in:home` operator from what actually gets sent to servers, or something. Really unsure. But worth a #frontend discussion to help think through it together.

Reply:
Opened discussion in https://chat.zulip.org/#narrow/channel/378-api-design/topic/in-home.20filter.20definition and pinged Alya so that it is easy to get feedback from everyone and add an api definition for the `in:home` filter.